lockc/src/bpf/lockc.bpf.c

```diff
@@ -390,6 +390,68 @@ int BPF_PROG(mount_audit, const char *dev_name, const struct path *path,
 	return ret;
 }
 
+/*
+ * setuid_audit - LSM program triggered when user UID is changed.
+ * The goal is to deny changing user ID from regular account to root user account.
+ * @cred *new: data structure with new user context
+ * @cred *old: data structure with old user context
+ * @flags: additional flags
+ * @ret_prev: return code of a previous BPF program using the sb_mount hook
+ *
+ * Return: 0 if changing UID is allowed. -EPERM if root account not allowed. -EFAULT if there was
+ * a problem with reading the kernel strings into buffers or any important
+ * buffer is NULL.
+ */
+SEC("lsm/task_fix_setuid")
+int BPF_PROG(setuid_audit, struct cred *new, const struct cred *old, int flags, int ret_prev)
+{
+	int ret = 0;
+	char comm[TASK_COMM_LEN];
+
+	pid_t pid = bpf_get_current_pid_tgid() >> 32;
+	enum container_policy_level policy_level = get_policy_level(pid);
+
+	if (bpf_get_current_comm(&comm, sizeof(comm)) < 0)
+		return -EFAULT;
+
+	bpf_printk("DEBUG: setuid process: %s\n", comm);
+
+	uid_t uid_old = BPF_CORE_READ(old, uid).val;
+	bpf_printk("DEBUG: setuid old: %d\n", uid_old);
+
+	uid_t uid_new = BPF_CORE_READ(new, uid).val;
+	bpf_printk("DEBUG: setuid new: %d\n", uid_new);
+
+	switch (policy_level) {
+	case POLICY_LEVEL_LOOKUP_ERR:
+		ret = -EPERM;
+		goto out;
+	case POLICY_LEVEL_NOT_FOUND:
+    bpf_printk("DEBUG: policy: not_found\n");
+		break;
```

Comment:
`POLICY_LEVEL_NOT_FOUND` means that it's a normal non-containerized process (I probably need to change that name to something which will make it clear.

So in that case, we don't want to perform any checks. but just allow the action (or handle `ret_prev`).

Therefore:

```suggestion
		goto out;
```

Reply:
I agree, done